src/main/java/com/upserve/uppend/AppendStorePartition.java

```diff
@@ -90,8 +109,11 @@ void append(String key, byte[] blob) {
         LookupKey lookupKey = new LookupKey(key);
         final int hash = keyHash(lookupKey);
 
-        return blocks.values(lookups[hash].getValue(lookupKey)).parallel().mapToObj(blobs[hash]::read);
-        // Consider sorting by blob pos or even grouping by the page of the blob pos and then flat-mapping the reads by page.
+        // Values stream can now be parallel, but it breaks everything...
+        LongStream longStream = blocks.values(lookups[hash].getValue(lookupKey));
+        return longStream.mapToObj(blobs[hash]::read);
```

Comment:
Applying Parallel here means the calling application cannot also be using parallel.

The previous version, using lazy concat did not suffer this issue because calling parallel on a stream is more of an aspirational statement than a demand.

The result is that our reads have never been parallel and I am unsure how to resolve this. Previously, when we hit this issue with using parallel during write we just removed that code - having decided it was not necessary (it was validating the page file metadata on open - that check and the metadata are now gone).

I think we should look at this in the larger context of how to make the gastronomer list queries actually execute in parallel.

Reply:
At present, some portions of the query are, but many are not parallel.